crypto/stack/stack.c

```diff
@@ -118,32 +118,59 @@ OPENSSL_STACK *OPENSSL_sk_new(OPENSSL_sk_compfunc c)
     return (NULL);
 }
 
-int OPENSSL_sk_insert(OPENSSL_STACK *st, const void *data, int loc)
+int OPENSSL_sk_reserve(OPENSSL_STACK *st, int n)
 {
-    if (st == NULL || st->num < 0 || st->num == INT_MAX) {
+    size_t target;
+    size_t num_alloc = st->num_alloc;
+    size_t sz;
+
+    if (st == NULL || st->num < 0)
         return 0;
-    }
 
-    if (st->num_alloc <= (size_t)(st->num + 1)) {
-        size_t doub_num_alloc = st->num_alloc * 2;
-        const char **tmpdata;
+    if (n <= 0)
+        return 1;
+
+    /* Sanity checks */
+    if (n > INT_MAX - st->num
+        || (size_t)n > SIZE_MAX / sizeof(char *) - st->num)
+        return 0;
+
+    target = st->num + n * sizeof(char *);
```

Comment:
Why n is multiplied by `sizeof(char*)`?

Reply:
Gone now.  Left over from the size_t based expansion's earlier code.
Although it was probably wrong then too :(